.travis.yml

```diff
@@ -26,7 +26,11 @@ before_script:
 script:
   - npm run lint:wrapper
   - npm run lint:contract-tests
-  - npm run test
+  - npm run test:token
+  - npm run test:contributor
+  - npm run test:contribution
+  - npm run test:proposal
+  - npm run test:kit
```

Comment:
This should be the same as the test script itself, because `&&` is supposed to exit with 1, if any of the commands on the way exit with 1.

Reply:
yeah true, I changed that also to 
```env:
  - TASK=lint:wrapper
  - TASK=lint:contract-tests
  - TASK=test:token
  - TASK=test:contributor
  - TASK=test:contribution
  - TASK=test:proposal

before_script:
  - npm run devchain > /dev/null &
  - sleep 5

script:
  - travis_wait 60 npm run $TASK
```
It is the same also....
But the thing is there is an error in Contribution tests, but it doesn't exit with 1!

Reply:
I don't see how the env vars make it clearer. I think it's actually less understandable that way. In fact, I wouldn't even know off the top of my head what happens when you assign the same env variable twice.

>  But the thing is there is an error in Contribution tests, but it doesn't exit with 1!

Yes, that's the thing that needs to be fixed. Not how the tests are run. The test task should just be `npm test` and nothing else.

Reply:
What I meant by the env vars make it clearer, in Travis u don't need to look at the whole log in case a test fail...
![1](https://user-images.githubusercontent.com/17862704/65396697-eb146900-dda0-11e9-9186-2184cbf79106.png)

Reply:
Oh, I see. The feature you're looking for is the build matrix:

https://docs.travis-ci.com/user/customizing-the-build/#naming-jobs-within-matrices